src/scheduler/job_info.c

```diff
@@ -2508,36 +2498,19 @@ create_resresv_sets(status *policy, server_info *sinfo)
 				free_resresv_set_array(rsets);
 				return NULL;
 			}
-			cur_rset->resresv_arr = malloc((len + 1) * sizeof(resource_resv *));
-			if (cur_rset->resresv_arr == NULL) {
-				log_err(errno, __func__, MEM_ERR_MSG);
-				free_resresv_set_array(rsets);
-				free_resresv_set(cur_rset);
-				return NULL;
-			}
 			cur_ind = j;
 			rsets[j++] = cur_rset;
 			rsets[j] = NULL;
 		} else
 			cur_rset = rsets[cur_ind];
 
-		cur_rset->resresv_arr[cur_rset->num_resresvs] = resresvs[i];
-		cur_rset->resresv_arr[++cur_rset->num_resresvs] = NULL;
 		resresvs[i]->ec_index = cur_ind;
 	}
 
-	/* tidy up */
-	for (i = 0; rsets[i] != NULL; i++) {
-		resource_resv **tmp_arr;
-		tmp_arr = realloc(rsets[i]->resresv_arr, (rsets[i]->num_resresvs + 1) * sizeof(resource_resv *));
-		if (tmp_arr != NULL)
-			rsets[i]->resresv_arr = tmp_arr;
-	}
-
 	tmp_rset_arr = realloc(rsets,(j + 1) * sizeof(resresv_set *));
 	if (tmp_rset_arr != NULL)
 		rsets = tmp_rset_arr;
-
+	i = count_array((void **)rsets);
```

Comment:
Since this is no longer an index variable, consider changing the name of it.

Reply:
Well, this variable is used to iterate through all jobs in the loop above this code. I am just reusing this variable to get the number of sets created.

Reply:
I'd still use a different variable.  Something like num_rsets.